### PR TITLE
Add compass button tests

### PR DESCRIPTION
**Pull request by VehpuS**
Adds tests for pressing / dragging the compass button in the NavigationControl. Part of the work on https://github.com/maplibre/maplibre-gl-js/pull/1852

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Write tests for all new functionality.

### CODE REVIEW

**Comment by VehpuS** [comment]
The build seems to have failed due something unrelated, unless I misunderstand the error...

File: src/util/browser.ts [hunk 1]
```diff
@@ -44,7 +44,11 @@ const exported = {
     hardwareConcurrency: typeof navigator !== 'undefined' && navigator.hardwareConcurrency || 4,
 
     get prefersReducedMotion(): boolean {
-        if (!matchMedia) return false;
+        try {
```
Comment (HarelM) [inline src/util/browser.ts:47]: Why was this added?

**Comment by HarelM** [comment]
Thanks! Looks great!

File: src/util/browser.ts (hunk 1 above)
Comment (VehpuS) [reply src/util/browser.ts:47]: Before this change I got an error that seemed to be exactly the reason for this function to exists: `ReferenceError: matchMedia is not defined`. I honestly planned to return to document / find a nicer solution but forgot 🤦. Having searched for a better solution just now, I see there's a solution already implemented in the testing utils (`setMatchMedia`) - I'll use it and push a correction.

**Review by HarelM — APPROVED** [review]
